Duplicate entries into player_stats

[2020-11-16T02:32:27.282] [ERROR] unhandledRejection - Unhandled Rejection at: Promise [object Promise]. Reason: Error: insert into `player_stats` (`first_play`, `last_active`, `player_id`) values ('2020-11-16 02:32:27', '2020-11-16 02:32:27', '287570174100635648') - ER_DUP_ENTRY: Duplicate entry '287570174100635648' for key 'player_stats_player_id_unique'. Trace: Error: ER_DUP_ENTRY: Duplicate entry 'x' for key 'player_stats_player_id_unique'

--- a/src/models/game_session.ts
+++ b/src/models/game_session.ts
@@ -99,11 +99,12 @@ export default class GameSession {
         }
 
         for (const participant of this.participants) {
-            this.incrementPlayerGamesPlayed(participant);
-        }
-
-        for (const playerScore of this.scoreboard.getPlayerScores()) {
-            this.incrementPlayerSongsGuessed(playerScore.id, playerScore.score);
+            await this.ensurePlayerStat(participant);
+            await this.incrementPlayerGamesPlayed(participant);
+            const playerScore = this.scoreboard.getPlayerScore(participant);
+            if (playerScore > 0) {
+                await this.incrementPlayerSongsGuessed(participant, playerScore);
+            }
         }
 
         await dbContext.kmq("guild_preferences")
@@ -325,7 +326,6 @@ export default class GameSession {
     }
 
     async incrementPlayerSongsGuessed(userId: string, score: number) {
-        await this.ensurePlayerStat(userId);
         await dbContext.kmq("player_stats")
             .where("player_id", "=", userId)
             .increment("songs_guessed", score)
@@ -335,7 +335,6 @@ export default class GameSession {
     }
 
     async incrementPlayerGamesPlayed(userId: string) {
-        await this.ensurePlayerStat(userId);
         await dbContext.kmq("player_stats")
             .where("player_id", "=", userId)
             .increment("games_played", 1);

